sdk/messaging/azeventhubs/internal/cbs.go

```diff
@@ -35,7 +35,7 @@ func NegotiateClaim(ctx context.Context, audience string, conn amqpwrap.AMQPClie
 	}
 
 	defer func() {
-		if err := link.Close(ctx); err != nil {
+		if err := link.Close(context.Background()); err != nil {
```

Comment:
Should these have a timeout, or does this happen on a separate goroutine?

Reply:
(just updating with offline conversation)

We're going to leave this as is, and double down and add this to more places. The proper cleanup and closing of links is important enough for the "health" of our connection that leaving it in an incomplete state can cause subtle issues that'd be best to avoid.